### ISSUE TITLE
multilabel classification using xgboost

### ISSUE THREAD

**Issue by longzaitianguo** [dmlc/xgboost#2454]
Dear all,

I am using xgboost and scikit-learn for a multilabel classification task as follows:
OneVsRestClassifier(XGBClassifier(...scale_pos_weight = ?))
However, since my dataset is very unbalanced, I need to set a scale_pos_weight value for each label. 
Then the question is, how can I set a separate scale_pos_weight for each XGBClassifier in OneVsRestClassifier.

Thank so much!

**Comment by pommedeterresautee** [comment]
As multiclass on XGBoost is just one versus all, you can separate each binary classifier and set the reweighting per classifier. Usually # negative / # positives is a good ratio.

**Comment by longzaitianguo** [comment]
@pommedeterresautee 
How to do this? As you see, the XGBClassifiers are embedded in the OneVsRestClassifier api. In addition, OneVsRestClassifier does not accept a classifier list. So I cannot figure out how to set a separate weight for each classifier. Or I need to build the classifier for each label and then merge them using the onevsall strategy manually?

**Comment by pommedeterresautee** [comment]
You can just simulate the multiclassifier yourself in a 1 vs all setting -> in this case you have several binary classifiers (that's what is done internally).

**Comment by pommedeterresautee** [comment]
https://github.com/dmlc/xgboost/issues/2087
